### DISCUSSION TITLE
Rhino 8 .NET 7 upgrade?

### DISCUSSION THREAD

**Discussion by lin-ycv** [visose/Robots#266] [Q&A] [answered]
Hi Vicente @visose 

I'm wondering if there's any plan to upgrade to .NET7 for Rhino 8?
There's no error with robots and Rhino 8, I'm just curious.

**Comment by visose** [comment, marked as answer]
Hi, nothings needs to change in the project to use .NET 7 in Rhino 8.
Targeting .NET 7 specifically instead of .NET Framework 4.8 does not make much difference in how the dll is compiled. It's only useful if you are going to use newer APIs that don't exist in older versions (and viceversa). I won't be doing that yet to keep it compatible with Rhino 7. Same as targeting RhinoCommon 8 vs 7.

**Comment by visose** [comment]
On the other hand I did apply some new C# 12 features recently, see: https://github.com/visose/Robots/commit/9e05939cc457257c667d02f1f0196bc45f4ebec9
Some are purely syntatic sugar but others help compile to more efficient IL (in practice I really doubt there will be any noticable difference).
